src/services/orderbook_service.ts

```diff
@@ -111,49 +111,15 @@ export class OrderBookService {
         const signedOrderEntities = (await this._connection.manager.find(SignedOrderEntity, {
             where: filterObject,
         })) as Array<Required<SignedOrderEntity>>;
-        let apiOrders = _.map(signedOrderEntities, orderUtils.deserializeOrderToAPIOrder);
+        const apiOrders = _.map(signedOrderEntities, orderUtils.deserializeOrderToAPIOrder);
         // Post-filters
-        apiOrders = apiOrders
-            .filter(
-                // traderAddress
-                apiOrder =>
-                    ordersFilterParams.traderAddress === undefined ||
-                    apiOrder.order.makerAddress === ordersFilterParams.traderAddress ||
-                    apiOrder.order.takerAddress === ordersFilterParams.traderAddress,
-            )
-            .filter(
-                // makerAssetAddress
-                apiOrder =>
-                    ordersFilterParams.makerAssetAddress === undefined ||
-                    orderUtils.includesTokenAddress(
-                        apiOrder.order.makerAssetData,
-                        ordersFilterParams.makerAssetAddress,
-                    ),
-            )
-            .filter(
-                // takerAssetAddress
-                apiOrder =>
-                    ordersFilterParams.takerAssetAddress === undefined ||
-                    orderUtils.includesTokenAddress(
-                        apiOrder.order.takerAssetData,
-                        ordersFilterParams.takerAssetAddress,
-                    ),
-            )
-            .filter(
-                // makerAssetProxyId
-                apiOrder =>
-                    ordersFilterParams.makerAssetProxyId === undefined ||
-                    assetDataUtils.decodeAssetDataOrThrow(apiOrder.order.makerAssetData).assetProxyId ===
-                        ordersFilterParams.makerAssetProxyId,
-            )
-            .filter(
-                // takerAssetProxyId
-                apiOrder =>
-                    ordersFilterParams.takerAssetProxyId === undefined ||
-                    assetDataUtils.decodeAssetDataOrThrow(apiOrder.order.takerAssetData).assetProxyId ===
-                        ordersFilterParams.takerAssetProxyId,
-            );
-        const paginatedApiOrders = paginationUtils.paginate(apiOrders, page, perPage);
+        const filteredApiOrders = orderUtils.filterOrders(apiOrders, ordersFilterParams);
+        // Remove expired orders
+        const dateNowSeconds = Date.now() / ONE_SECOND_MS;
+        const freshFilteredApiOrders = filteredApiOrders.filter(apiOrder =>
```

Comment:
Why doesn't `Remove expired orders` go into `orderUtils.filterOrders`? is that because it's a temporary fix?

Reply:
`orderUtils.filterOrders` works on the filters specified in the SRA, which `expirationTimeSeconds` is not part of.